src/createUniqueTag.ts

```diff
@@ -0,0 +1,70 @@
+/**
+ * Global counter to scope the custom elements
+ *
+ * @type {number}
+ */
+let counter: number = Math.round(Math.random() * 100000);
+
+/**
+ * Allowed tag name chars
+ *
+ * @type {string}
+ */
+const chars: string = `-|\\.|[0-9]|[a-z]`;
+
+/**
+ * Regular expression to check if a value is a valid tag name
+ *
+ * @type {RegExp}
+ */
+const tagRegExp: RegExp = new RegExp(`[a-z](${chars})*-(${chars})*`);
+
+/**
+ * Checks if the tag name is valid
+ *
+ * @param {string} tag
+ * @returns {boolean}
+ */
+const isValid = (tag: string): boolean => tagRegExp.exec(tag) !== null;
```

Comment:
I believe it's appropriate to not have to check whether the tag is valid, since the custom element registry will provide a detailed error message instead 
![image](https://user-images.githubusercontent.com/12037530/86632232-f75e6100-bf83-11ea-84a8-5ec7c2fa5138.png)